service/scaladsl/server/src/main/scala/com/lightbend/lagom/scaladsl/server/LagomApplicationLoader.scala

```diff
@@ -228,11 +228,15 @@ abstract class LagomApplication(context: LagomApplicationContext)
 
   override val httpFilters: Seq[EssentialFilter] = Nil
 
-  override lazy val configuration: Configuration = Configuration.load(environment) ++
-    context.playContext.initialConfiguration ++ additionalConfiguration.configuration
+  override lazy val configuration: Configuration = {
+    val additionalConfig = new Configuration(additionalConfiguration.configuration)
+    Configuration.load(environment) ++ context.playContext.initialConfiguration ++ additionalConfig
+  }
+
+  lazy val underlyingConfig: Config = configuration.underlying
```

Comment:
This one actually can't be called config... :( `LagomApplicationLoader` extends a class which, at some point, has a member called `config`.